pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -129,65 +122,65 @@ var _ = Describe("Custom Constraints", func() {
 		})
 		It("should schedule pods that have matching preferences", func() {
 			provisioner.Spec.Labels = map[string]string{"test-key": "test-value"}
-			pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod(
+			pod := ExpectProvisioned(ctx, env.Client, controller, provisioner, test.UnschedulablePod(
 				test.PodOptions{NodePreferences: []v1.NodeSelectorRequirement{
 					{Key: "test-key", Operator: v1.NodeSelectorOpIn, Values: []string{"another-value", "test-value"}},
 				}},
 			))[0]
 			node := ExpectScheduled(ctx, env.Client, pod)
 			Expect(node.Labels).To(HaveKeyWithValue("test-key", "test-value"))
 		})
-		It("should not schedule pods with have conflicting preferences", func() {
+		It("should schedule pods with have conflicting preferences", func() {
 			provisioner.Spec.Labels = map[string]string{"test-key": "test-value"}
-			pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod(
+			pod := ExpectProvisioned(ctx, env.Client, controller, provisioner, test.UnschedulablePod(
 				test.PodOptions{NodePreferences: []v1.NodeSelectorRequirement{
 					{Key: "test-key", Operator: v1.NodeSelectorOpNotIn, Values: []string{"test-value"}},
 				}},
 			))[0]
-			ExpectNotScheduled(ctx, env.Client, pod)
+			ExpectScheduled(ctx, env.Client, pod)
```

Comment:
Why does this schedule now?   There are a few tests like this where it looks like just the condition was changed.

Reply:
Previously preferences needed to execute twice and the test only executed once.